test/xml/test_schema.rb

```diff
@@ -171,6 +171,10 @@ class TestNokogiriXMLSchema < Nokogiri::TestCase
 
         assert(errors = xsd.validate(tempfile.path))
         assert_equal(2, errors.length)
+        assert_equal(
```

Comment:
In this particular test case, `error->node` looks like it's `NULL` (on my system, anyway).

You may want to write a new test case that's crafted so that `error->node` is present (and honestly I'm not sure when it's present or not, I'd have to dig into the libxml2 code). Let me know if you need assistance.

Reply:
Interesting. I'm running two ways: `bundle exec rake test` (whole suite) and `bundle exec rake test A=-n/validate_invalid_file/` (individual test).

Reply:
Are you running with the packaged libxml2 or are you using your system library?

Reply:
never mind... needs different test

Reply:
I'm betting you're running with system libxml2 old enough that it doesn't have this commit in it yet: https://github.com/GNOME/libxml2/commit/e20c9c148c725e2933efa143ee6a543a5cae4204

(looks like 2.9.10 or earlier)

Reply:
I just put the test in the wrong place, it needed to go in the one above. it was failing locally too

Reply:
It appears errors only render when validating a document and not a file. I am guessing when validating a document a node exists where as a file there is no node.